/*

   calin/proto/diagnostics/value_capture.proto -- Stephen Fegan -- 2016-05-23

   Protobufs for value capture

   Copyright 2016, Stephen Fegan <sfegan@llr.in2p3.fr>
   LLR, Ecole Polytechnique, CNRS/IN2P3

   This file is part of "calin"

   "calin" is free software: you can redistribute it and/or modify it
   under the terms of the GNU General Public License version 2 or
   later, as published by the Free Software Foundation.

   "calin" is distributed in the hope that it will be useful, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   General Public License for more details.

*/

syntax = "proto3";

import "calin.proto";
//import "math/histogram.proto";
//import "common_types.proto";

package calin.ix.diagnostics.value_capture;

message CapturedInt32Values {
  int32 guard_value                                        = 1 [
    (CFO).desc = "Guard value indicating that no value was captured." ];
  repeated int32 value                                     = 2 [
    (CFO).desc = "Captured values." ];
};

message CapturedInt64Values {
  int64 guard_value                                        = 1 [
    (CFO).desc = "Guard value indicating that no value was captured." ];
  repeated int64 value                                     = 2 [
    (CFO).desc = "Captured values." ];
};

message CapturedDoubleValues {
  double guard_value                                       = 1 [
    (CFO).desc = "Guard value indicating that no value was captured." ];
  repeated double value                                    = 2 [
    (CFO).desc = "Captured values." ];
};
